### PR TITLE
(#1166) Remove some more non OO matchers

### PR DESCRIPTION
**Pull request by victornoel**
This is for #1166

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#1197](https://github.com/yegor256/cactoos/issues/1197) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#1197](https://github.com/yegor256/cactoos/issues/1197) is assigned to @fabriciofx[/z](https://www.0crat.com/u/fabriciofx), here is [why](https://www.0crat.com/footprint/C63314D6Z/2f47ea26-3b58-4d71-8780-57b4de3356a1); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @paulodamaso[/z](https://www.0crat.com/u/paulodamaso) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be no monetary reward for this job

File: src/test/java/org/cactoos/iterator/IteratorOfTest.java [hunk 1]
```diff
@@ -39,45 +42,54 @@
...
+        new Assertion<>(
+            "must throw an exception if empty",
+            () -> new IteratorOf<>().next(),
+            new Throws<>(NoSuchElementException.class)
+        ).affirm();
     }
 
     @Test
     public void nonEmptyIteratorDoesNotHaveNext() {
         final IteratorOf<Integer> iterator = this.iteratorWithFetchedElements();
```
Comment (fabriciofx) [inline src/test/java/org/cactoos/iterator/IteratorOfTest.java]: @victornoel According with [this Yegor's post](https://www.yegor256.com/2016/05/03/test-methods-must-share-nothing.html) tests must share nothing. So, can you leave a puzzle to refactor this? (it's interesting leave the link of the post too).

**Comment by fabriciofx** [comment]
@victornoel ping

**Comment by victornoel** [comment]
@fabriciofx it is done

**Comment by fabriciofx** [comment]
@victornoel please, fix the puzzle's text, according to Travis/CI error message.

**Comment by victornoel** [comment]
@fabriciofx it's done, sorry about that

**Comment by victornoel** [comment]
@fabriciofx ping?

**Comment by paulodamaso** [comment]
@fabriciofx please go on

**Comment by paulodamaso** [comment]
@0crat assign me

**Comment by paulodamaso** [comment]
@victornoel Seems good to me, let's merge it

**Comment by paulodamaso** [comment]
@rultor merge

**Comment by 0crat** [comment]
> @0crat assign me ([here](https://github.com/yegor256/cactoos/issues/1197#issuecomment-546909450))

@paulodamaso Job [#1197](https://github.com/yegor256/cactoos/issues/1197) is closed, can't start order

**Comment by 0crat** [comment]
@fabriciofx[/z](https://www.0crat.com/u/fabriciofx) resigned from [#1197](https://github.com/yegor256/cactoos/issues/1197), since the job is not in scope anymore

**Comment by 0crat** [comment]
The job [#1197](https://github.com/yegor256/cactoos/issues/1197) is now out of scope

**Comment by 0crat** [comment]
There is an unrecoverable failure on my side. Please, submit it [here](https://github.com/zerocracy/farm/issues):

```
PID: 12@b7e726625068, thread: PQ-C63314D6Z
com.zerocracy.claims.ClaimIn[214] java.lang.IllegalArgumentException: Parameter "age" not found in "Order was canceled" among: [reason, priority, job, cause, _parent_sig, flow, voluntarily, login]
```

[0.50.34](https://github.com/zerocracy/farm): CID: [dac70b99-61f0-429a-9dbf-d7174467bc47](https://www.0crat.com/footprint/C63314D6Z/dac70b99-61f0-429a-9dbf-d7174467bc47), Type: "Order was canceled"

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso)

**Pull request merged**